docs/docs/concepts/Sync_Async.md

```diff
@@ -0,0 +1,204 @@
+    });
+
+    return count;
+});
+```
+
+The above code works as expected. It first checks the cache and if the cache exists it's returned. Otherwise it will perform the calculation and update the cache. But the above example is not optimal. The callback is a coroutine, which means that it will always be calculated on a background thread. Intrinsically, moving to a background thread is costly. The best approach would be to check the cache synchronously and, only if its not there, move to the background. Triggers and Functions provides for starting synchronously and then moving asynchronously using `executeAsync` function as required. The new code:
+
+```js
+#!js api_version=1.0 name=lib
+
+redis.registerAsyncFunction('test', function(client, expected_name){
+    // check the cache first
+    var cached_value = client.call('get', expected_name + '_count');
+    if (cached_value != null) {
+        return cached_value;
+    }
+
+    // cache is not set, move to background
+    return client.executeAsync(async function(async_client) {
+        var count = 0;
+        var cursor = '0';
+        do{
+            async_client.block((client)=>{
+                var res = client.call('scan', cursor);
+                cursor = res[0];
+                var keys = res[1];
+                keys.forEach((key) => {
+                    if (client.call('hget', key, 'name') == expected_name) {
+                        count += 1;
+                    }
+                });
+            });
+        } while(cursor != '0');
+
+        // set count to the cache wil 5 seconds expiration
+        async_client.block((client)=>{
+            client.call('set', expected_name + '_count', count);
+            client.call('expire', expected_name + '_count', 5);
+        });
+
+        return count;
+    });
+});
+```
+
+`executeAsync` will return a `Promise` object. When Triggers and Functions sees that the function returns a Promise, it waits for the promise to be resolved and returns its result to the client. The above implementation will be much faster in the case of cache hit.
+
+**Notice** that even though we registered a synchronous function (not a coroutine) we still used `registerAsyncFunction`. This is because our function has the potential of blocking the client, taking the execution to the background. If we had used `registerFunction`, Triggers and Functions would not have allowed the function to block the client and it would have ignored the returned promise object.
+
+**Also notice** it is not always possible to wait for a promise to be resolved. If the command is called inside a `multi/exec` it is not possible to block it and wait for the promise. In such cases the client will get an error. It is possible to check if blocking the client is allowed using the `client.isBlockAllowed()` function, which will return `true` if it is OK to wait for a promise to be resolved and `false` if it is not possible.
+
+
```

Comment:
What about this: https://github.com/RedisGears/RedisGears/blob/v2.0.7-m08/docs/sync_and_async_run.md#call-blocking-commands
Please go over the docs and make sure you are not missing anything that was added after your branch was created.